examples/object_detection/main.py

```diff
@@ -290,7 +290,10 @@ def train(net, compression_ctrl, train_data_loader, test_data_loader, criterion,
     best_compression_level = CompressionLevel.NONE
     test_freq_in_epochs = max(config.test_interval // epoch_size, 1)
 
+    last_epoch = -1 if config.start_iter == 0 else config.start_iter // epoch_size
+    compression_ctrl.scheduler.epoch_step(last_epoch)
     for iteration in range(config.start_iter, config['max_iter']):
+        compression_ctrl.scheduler.step(iteration - config.start_iter)
```

Comment:
Shouldn't it be this way (without other calls of epoch_step):
```
epoch = iteration // epoch_size
if iteration % epoch_size == 0:
    compression_ctrl.scheduler.epoch_step(epoch) # or without epoch, as for resuming we've already loaded scheduler state from checkpoint
compression_ctrl.scheduler.step(iteration) # or without iteration
```
It also looks like currently it won't work correctly for resuming from intermediate step (i.e. in a middle of epoch) with per-step scheduler. This refers to scheduler implementation rather than to this sample.

Reply:
Done.